### DISCUSSION TITLE
Handling multiple queries based on user selection

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3084] [Q&A]
We are trying to integrate a multi select with react query to fetch data based on the multi selection of an item.
Each item will be fetched from a single api endpoint '../api/${itemId}

Can we pass the item selection state as a function key ? to run multiple queries ?
Shouold we use useQueries or useQuery ?

Any input appreciated.

**Comment by TkDodo** [comment]
Sounds like a good candidate for useQueries to me.

**Reply by cubecleveland** [reply]
Thanks!

how would i hav run the multiple queries based on the state selection. i have a state that has an array of the selction ids 
let selctionState = [id1, id2, id3, id4];
And the query to fetch is fetchQuery(id)
Can u post and example of how to form this using useQueries ?

**Reply by TkDodo** [reply]
```
useQueries(selectionState.map(id => ({
  queryKey: ['something', id],
  queryFn: () => fetchQuery(id)
})
```
